pennylane/circuit_graph.py

```diff
@@ -218,22 +193,14 @@ def serialize(self):
         """
         serialization_string = ""
         delimiter = "!"
-        variable_delimiter = "V"
 
         for op in self.operations_in_order:
             serialization_string += op.name
 
             for param in op.data:
-                if isinstance(param, Variable):
-                    serialization_string += delimiter
-                    serialization_string += variable_delimiter
-                    serialization_string += str(param.idx)
-                    serialization_string += delimiter
-
-                else:
-                    serialization_string += delimiter
-                    serialization_string += str(param)
-                    serialization_string += delimiter
+                serialization_string += delimiter
+                serialization_string += str(param)
+                serialization_string += delimiter
```

Comment:
Would the following two circuits produce a different serialization if `x` and `y` are free and `x !=y`? The serialization used to match as `x` and `y` would be described as `Variable` objects.
```python
def circuit1(x):
    qml.RX(x,wires=0)
    return qml.expval(qml.PauliZ(0))
```
```python
def circuit2(y):
    qml.RX(y,wires=0)
    return qml.expval(qml.PauliZ(0))
```
(This would be important because e.g., device caching depends on hashes that are produced from serialized circuits.)

Reply:
Yep, 💯 , this is something I was worried about as well. It might be worth thinking about this in a follow-up PR.

However, this is likely not priority for now, as:

1. As far as I am aware, only the Forest plugin requires this functionality?
2. We are planning to overhaul and re-think QNode serialization in the short term.

There is one short term issue though - this might affect the forest demos in the QML repo.

Reply:
Yep, and in general the `QubitDevice` caching functionality also depends on it. Although now I'm wondering if we actually wanted to cache two circuits that were different only in their free parameter values on the device level 🤔 (Could this lead to bugs in the old core? 🤔) The new functionality would seem more fitting.

Reply:
> The new functionality would seem more fitting.

Yep, I would tend to agree 🤔